src/script/lua_api/l_object.cpp

```diff
@@ -1698,43 +1698,167 @@ int ObjectRef::l_hud_get_hotbar_selected_image(lua_State *L)
 
-	std::string type = luaL_checkstring(L, 3);
+		lua_getfield(L, 2, "type");
+		if (!lua_isnil(L, -1))
+			skybox_params.type = luaL_checkstring(L, -1);
+		lua_pop(L, 1);
 
-	std::vector<std::string> params;
-	if (lua_istable(L, 4)) {
-		lua_pushnil(L);
-		while (lua_next(L, 4) != 0) {
-			// key at index -2 and value at index -1
-			if (lua_isstring(L, -1))
-				params.emplace_back(readParam<std::string>(L, -1));
-			else
-				params.emplace_back("");
-			// removes value, keeps key for next iteration
+		lua_getfield(L, 2, "textures");
+		skybox_params.params.clear();
+		if (!lua_istable(L, -1)) {
+			//skybox_params.params.emplace_back("");
+		} else {
+			lua_pushnil(L);
+			while (lua_next(L, -2) != 0) {
+				// Key is at index -2 and value at index -1
+				skybox_params.params.emplace_back(readParam<std::string>(L, -1));
+				// Removes the value, but keeps the key for iteration
+				lua_pop(L, 1);
+			}
+		}
+		lua_pop(L, 1);
+
+		/*
+		We want to avoid crashes, so we're checking even if we're not using them.
+		However, we want to ensure that the skybox can be set to nil when
+		using "regular" or "plain" skybox modes as textures aren't needed.
+		*/
+
+		if (skybox_params.params.size() != 6 && skybox_params.params.size() > 0)
+			throw LuaError("Skybox expects 6 textures!");
+
+		skybox_params.clouds = getboolfield_default(L, 2,
+			"clouds", skybox_params.clouds);
+
+		lua_getfield(L, 2, "sky_color");	
+		if (lua_istable(L, -1)) {
+			lua_getfield(L, 3, "day_sky");
+			if (!lua_isnil(L, -1))
+				read_color(L, -1, &skybox_params.day_sky);
+			lua_pop(L, 1);
+
+			lua_getfield(L, 3, "day_horizon");
+			if (!lua_isnil(L, -1))
+				read_color(L, -1, &skybox_params.day_horizon);
+			lua_pop(L, 1);
+
+			lua_getfield(L, 3, "dawn_sky");
+			if (!lua_isnil(L, -1))
+				read_color(L, -1, &skybox_params.dawn_sky);
+			lua_pop(L, 1);
+
+			lua_getfield(L, 3, "dawn_horizon");
+			if (!lua_isnil(L, -1))
+				read_color(L, -1, &skybox_params.dawn_horizon);
 			lua_pop(L, 1);
+
+			lua_getfield(L, 3, "night_sky");
+			if (!lua_isnil(L, -1))
+				read_color(L, -1, &skybox_params.night_sky);
+			lua_pop(L, 1);
+
+			lua_getfield(L, 3, "night_horizon");
+			if (!lua_isnil(L, -1))
+				read_color(L, -1, &skybox_params.night_horizon);
+			lua_pop(L, 1);
+
+			lua_getfield(L, 3, "indoors");
+			if (!lua_isnil(L, -1))
+				read_color(L, -1, &skybox_params.indoors);
+			lua_pop(L, 1);
+
+			lua_getfield(L, 3, "sun_tint");
+			if (!lua_isnil(L, -1))
+				read_color(L, -1, &skybox_params.sun_tint);
+			lua_pop(L, 1);
+
+			lua_getfield(L, 3, "moon_tint");
+			if (!lua_isnil(L, -1))
+				read_color(L, -1, &skybox_params.moon_tint);
+			lua_pop(L, 1);
+
+			lua_getfield(L, 3, "tint_type");
+			if (!lua_isnil(L, -1)) {
+				skybox_params.tint_type = luaL_checkstring(L, -1);
+				lua_pop(L, 1);
+			}
+
+			// Because we need to leave the "sky_color" table.
+			lua_pop(L, 1);
+			getServer(L)->setSky(player, skybox_params);
+			lua_pushboolean(L, true);
+		}
```

Comment:
Seems that ```getServer(L)->setSky``` and ```lua_pushboolean``` calls are misplaced. They should be outside this block so they are also called when no "```sky_color```" table given. I had to change that to make following Lua code work:
```lua
player:set_sky({base_color = "#f1f4ee", type = "skybox", clouds = false, textures = {"Up.jpg", "Down.jpg", "Front.jpg", "Back.jpg", "Left.jpg", "Right.jpg"} } )
```